e2e_tests/tests/task/test_generic_tasks.py

```diff
@@ -0,0 +1,281 @@
+import pathlib
+import subprocess
+import time
+
+import pytest
+
+from determined.cli import ntsc
+from determined.common import api, context, util
+from determined.common.api import bindings
+from tests import api_utils
+from tests import config as conf
+
+
+def wait_for_task_state(
+    test_session: api.Session,
+    task_id: str,
+    expected_state: bindings.v1GenericTaskState,
+    timeout: int,
+) -> bool:
+    t = 0
+    found = False
+    while t < timeout and not found:
+        resp = bindings.get_GetTask(test_session, taskId=task_id)
+        found = expected_state == resp.task.taskState
+        t += 1
+        time.sleep(1)
+    return found
```

Comment:
nit: time loops are easier to write and read if you think in terms of deadlines (absolute times) instead of timeouts (relative times):

```python
deadline = time.time() + timeout
while time.time() < deadline:
    resp = bindings.get_GetTask(...)
    if expected_state == resp.task.taskState:
         return True
    time.sleep(0.1)
return False
```